cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -1470,23 +1471,21 @@ impl ManagedEnvironment {
 
         // Check whether we can fast-forward the remote branch to the local branch,
         // if not the environment has diverged.
-        // if `--force` flag is set we skip this check
-        if !force {
-            let consistent_history = self
-                .floxmeta
-                .git
-                .branch_contains_commit(&project_branch, &sync_branch)
-                .map_err(ManagedEnvironmentError::Git)?;
-            if !consistent_history {
-                Err(ManagedEnvironmentError::Diverged)?;
-            }
+        let consistent_history = self
+            .floxmeta
+            .git
+            .branch_contains_commit(&project_branch, &sync_branch)
+            .map_err(ManagedEnvironmentError::Git)?;
+        if !consistent_history && !force {
+            Err(ManagedEnvironmentError::Diverged)?;
+        }
 
-            let sync_branch_commit = self.floxmeta.git.branch_hash(&sync_branch).ok();
-            let project_branch_commit = self.floxmeta.git.branch_hash(&project_branch).ok();
+        let sync_branch_commit = self.floxmeta.git.branch_hash(&sync_branch).ok();
+        let project_branch_commit = self.floxmeta.git.branch_hash(&project_branch).ok();
 
-            if sync_branch_commit == project_branch_commit {
-                return Ok(PullResult::UpToDate);
-            }
+        // Regardless of whether `--force` is set, we want to accurately return UpToDate
+        if sync_branch_commit == project_branch_commit {
+            return Ok(PullResult::UpToDate);
```

Comment:
This broke a test:
- https://github.com/flox/flox/blob/0b01155c5c5954debb58b9767d34595ccb3cd18d/cli/tests/environment-managed.bats#L447-L461
- https://github.com/flox/flox/actions/runs/11982165653/job/33409733471#step:8:188

Moving it after `reset_local_env_to_current_generation` looks like it will reset without building.